Mage.Sets/src/mage/cards/w/Wish.java

```diff
@@ -0,0 +1,104 @@
+package mage.cards.w;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AsThoughEffectType;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author weirddan455
+ */
+public final class Wish extends CardImpl {
+
+    public Wish(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{2}{R}");
+
+        // You may play a card you own from outside the game this turn.
+        this.getSpellAbility().addEffect(new WishEffect());
+    }
+
+    private Wish(final Wish card) {
+        super(card);
+    }
+
+    @Override
+    public Wish copy() {
+        return new Wish(this);
+    }
+}
+
+class WishEffect extends OneShotEffect {
+
+    public WishEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "You may play a card you own from outside the game this turn";
+    }
+
+    private WishEffect(final WishEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public WishEffect copy() {
+        return new WishEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller != null && !controller.getSideboard().isEmpty()) {
+            controller.lookAtCards(source, "Sideboard", controller.getSideboard(), game);
+            game.addEffect(new WishPlayFromSideboardEffect(), source);
+            return true;
+        }
+        return false;
+    }
+}
+
+class WishPlayFromSideboardEffect extends AsThoughEffectImpl {
+
+    public WishPlayFromSideboardEffect() {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.EndOfTurn, Outcome.Benefit);
+    }
+
+    private WishPlayFromSideboardEffect(final WishPlayFromSideboardEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public WishPlayFromSideboardEffect copy() {
+        return new WishPlayFromSideboardEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        if (source.getControllerId().equals(affectedControllerId)) {
+            Player controller = game.getPlayer(source.getControllerId());
+            UUID mainCardId = CardUtil.getMainCardId(game, objectId);
+            if (controller != null && controller.getSideboard().contains(mainCardId)) {
+                // Only allow one card to be played
+                if (!game.inCheckPlayableState()) {
+                    discard();
```

Comment:
I don't like that code. It's looks potentially buggy. `AsThoughEffect` can be called multiple times from any places by game engine (e.g. on playable checks, on AI simulations, on user's feedback clicks check, etc) -- so it can't be one time used.

Maybe there are another solution possible: 
- users clicks on the wish card
- see cards list from sideboard and select to cast;
- cast it as normal card by `player.cast(player.chooseAbilityForCast(...`
- save in `GameState` value that it already cast on that tune.

SproutbackTrudge example:
![shot_210715_201735](https://user-images.githubusercontent.com/8344157/125822207-7450cf3e-4122-4a38-a1da-55d4a6d94f28.png)

Reply:
That's not how the card is worded though.  You don't choose a card to cast as it's resolving.  It just lets you play any card from your sideboard this turn.

Reply:
And it lets you play lands so player.cast won't work anyway.